src/standalone/api/api.ts

```diff
@@ -133,10 +129,23 @@ export function buildApi(
             const dataViewerProviderService = serviceContainer.get<IDataViewerFactory>(IDataViewerFactory);
             await dataViewerProviderService.create(dataProvider, title);
         },
-        registerRemoteServerProvider(picker: IJupyterUriProvider): void {
+        registerRemoteServerProvider(provider: IJupyterUriProvider): IDisposable {
             sendApiUsageTelemetry(extensions, 'registerRemoteServerProvider');
             const container = serviceContainer.get<IJupyterUriProviderRegistration>(IJupyterUriProviderRegistration);
-            container.registerProvider(picker);
+            let disposeHook = noop;
+            const register = async () => {
+                const extensions = serviceContainer.get<IExtensions>(IExtensions);
+                const extensionId = provider.id.startsWith('_builtin')
+                    ? JVSC_EXTENSION_ID
+                    : (await extensions.determineExtensionFromCallStack()).extensionId;
+                container.registerProvider(Object.assign(provider, { extensionId }));
```

Comment:
`registerProvider` supports sending a disposable
I believe you added this, so that one has a way to remove the server provider.
Hence i changed the return value to now return a `Disposable`, that is a change in the API, but safe,